tests/benchmark/runner/src/createResultTable.ts

```diff
@@ -0,0 +1,204 @@
+import Map from '@dojo/shim/Map';
+import { endsWith } from '@dojo/shim/string';
+import { JSONResult, frameworks, FrameworkData } from './common';
+import { BenchmarkType, Benchmark, benchmarks } from './benchmarks';
+
+const dots = require('dot').process({
+	path: './'
+});
+
+let frameworkMap = new Map<string, FrameworkData>();
+frameworks.map(f => frameworkMap.set(f.name, f));
+
+let results: Map<string, Map<string, JSONResult>> = new Map();
+
+fs.readdirSync('./results').filter(file => endsWith(file, '.json')).forEach(name => {
+	let data = <JSONResult> JSON.parse(fs.readFileSync('./results/' + name, {
+		encoding: 'utf-8'
+	}));
+
+	if (!frameworkMap.has(data.framework)) {
+		console.log('WARN: No entry in commons.ts for ' + data.framework + '. Data will not appear in result table.');
+	} else {
+		const dataFrameworkResult = results.get(data.framework);
+
+		if (dataFrameworkResult) {
+			dataFrameworkResult.set(data.benchmark, data);
+		} else {
+			results.set(data.framework, new Map());
+		}
+	}
+});
+
+let cpuBenchmarks = benchmarks.filter(benchmark => benchmark.type === BenchmarkType.CPU || benchmark.type === BenchmarkType.STARTUP);
+let memBenchmarks = benchmarks.filter(benchmark => benchmark.type === BenchmarkType.MEM);
+let cpuBenchmarkCount = cpuBenchmarks.length;
+
+let getValue = (framework: string, benchmark: string) => {
+	const frameworkResult = results.get(framework);
+
+	if (frameworkResult) {
+		return frameworkResult.get(benchmark);
+	}
+};
+
+function color(factor: number): string {
+	if (factor < 2.0) {
+		let a = (factor - 1.0);
+		let r = (1.0 - a) * 99 + a * 255;
+		let g = (1.0 - a) * 191 + a * 236;
+		let b = (1.0 - a) * 124 + a * 132;
+		return `rgb(${r.toFixed(0)}, ${g.toFixed(0)}, ${b.toFixed(0)})`;
+	} else  {
+		let a = Math.min((factor - 2.0) / 2.0, 1.0);
+		let r = (1.0 - a) * 255 + a * 249;
+		let g = (1.0 - a) * 236 + a * 105;
+		let b = (1.0 - a) * 132 + a * 108;
+		return `rgb(${r.toFixed(0)}, ${g.toFixed(0)}, ${b.toFixed(0)})`;
+	}
+}
+
+interface TestData {
+	mean: string;
+	deviation: string;
+	factor: string;
+	styleClass: string;
+}
+
+class BenchResultList {
+	tests: Array<TestData | null>;
+	name: string;
+	description: string;
+
+	constructor(benchmark: Benchmark) {
+		this.tests = [];
+		this.name = benchmark.label;
+		this.description = benchmark.description;
+	}
+}
+
+interface FrameworkPredicate {
+	(framework: FrameworkData): boolean;
+}
+
+let generateBenchData = (benchmarks: Array<Benchmark>, frameworkPredicate: FrameworkPredicate, referenceName: string) => {
+	let benches: Array<BenchResultList> = [];
+
+	let filteredFrameworks = frameworks.filter(f => frameworkPredicate(f)).slice();
+
+	let sortedFrameworks = filteredFrameworks.sort((a: FrameworkData, b: FrameworkData) => {
+		if (a.name === referenceName) {
+			if (b.name === referenceName) { return 0; }
+			console.log('found reference name', referenceName);
+			return 1;
+		} else if (b.name === referenceName) {
```

Comment:
a hangover from the original code, should we apply the dojo 2 style guide rules? like else on a new line

Reply:
I would recommend we don't use code review for applying style guide rules, specifically those which should be automated (so builds should fail is they do not match style guide rules)